test/e2e/windows/reboot_node.go

```diff
@@ -176,32 +177,24 @@ var _ = sigDescribe(feature.Windows, "[Excluded:WindowsDocker] [MinimumKubeletVe
 
 		restartCount := 0
 
-		timeout := time.After(time.Minute * 10)
-	FOR:
-		for {
-			select {
-			case <-timeout:
-				break FOR
-			default:
-				if restartCount > 0 {
-					break FOR
-				}
-				ginkgo.By("Then checking existed agn-test-pod is running on the rebooted host")
-				agnPodOut, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(ctx, agnPod.Name, metav1.GetOptions{})
-				if err == nil {
-					lastRestartCount := podutil.GetExistingContainerStatus(agnPodOut.Status.ContainerStatuses, "windows-container").RestartCount
-					restartCount = int(lastRestartCount - initialRestartCount)
-				}
-				time.Sleep(time.Second * 30)
+		ginkgo.By("Waiting for nodes to be rebooted")
+		gomega.Eventually(ctx, func(ctx context.Context) string {
+			refreshNode, err := f.ClientSet.CoreV1().Nodes().Get(ctx, targetNode.Name, metav1.GetOptions{})
+			if err != nil {
+				return ""
 			}
-		}
-
-		ginkgo.By("Checking whether agn-test-pod is rebooted")
-		gomega.Expect(restartCount).To(gomega.Equal(1), "restart count of agn-test-pod is 1")
+			return refreshNode.Status.NodeInfo.BootID
+		}).WithPolling(time.Second*30).WithTimeout(time.Minute*10).
+			ShouldNot(gomega.Equal(bootID), "node does not get rebooted")
```

Comment:
I believe the bootid is an incremented value on Windows (https://github.com/kubernetes/kubernetes/issues/108886#issuecomment-1599597663) so we could use to be more accurate?

```suggestion
			Should(BeNumerically(">", bootID)), "node was rebooted")
```

Reply:
should be "node was not rebooted"?